### PR TITLE
refactor: Refacto/simplify docker tag and push and exec

### PR DESCRIPTION
**Pull request by deimosfr**

### CODE REVIEW

File: src/cmd/utilities.rs [hunk 1]
```diff
@@ -58,36 +58,17 @@ where
     cmd
 }
 
-pub fn exec<P>(binary: P, args: Vec<&str>) -> Result<(), SimpleError>
+pub fn exec<P>(binary: P, args: Vec<&str>, envs: Option<Vec<(&str, &str)>>) -> Result<(), SimpleError>
```
Comment (erebe) [inline src/cmd/utilities.rs:61]: Change from Option<Vec> to Vec.
Pass am empty vector if you don't want env variables

File: src/container_registry/utilities.rs [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    dest: String,
+) -> Result<(), SimpleError> {
+    let image_with_tag = format!("{}:{}", image_name, image_tag);
+    let registry_provider = match container_registry_kind {
+        Kind::DockerHub => "DockerHub",
+        Kind::Ecr => "AWS ECR",
+        Kind::Docr => "DigitalOcean Registry",
+    };
+    let docker_environment_variables = match docker_envs {
+        None => vec![("", "")],
```
Comment (erebe) [inline src/container_registry/utilities.rs:22]: why not just vec![] ?

File: src/cmd/utilities.rs (hunk 1 above)
Comment (deimosfr) [reply src/cmd/utilities.rs:61]: I agree with you and tought about it as well but felt None is move explicit and easier than vec!["", ""]

File: src/container_registry/utilities.rs (hunk 2 above)
Comment (deimosfr) [reply src/container_registry/utilities.rs:22]: useless, therefore essential :)

You're right, removing them

**Review by erebe — APPROVED** [review]

File: src/cmd/utilities.rs [hunk 3]
```diff
@@ -49,45 +49,22 @@ where
...
-    }
+    envs.into_iter().for_each(|(k, v)| {
+        cmd.env(k, v);
+    });
 
     cmd
 }
 
-pub fn exec<P>(binary: P, args: Vec<&str>) -> Result<(), SimpleError>
+pub fn exec<P>(binary: P, args: Vec<&str>, envs: &Vec<(&str, &str)>) -> Result<(), SimpleError>
```
Comment (erebe) [inline src/cmd/utilities.rs:59]: missing this one

Comment (erebe) [inline src/cmd/utilities.rs:59]: missing this one

**Review by erebe — CHANGES_REQUESTED** [review]

File: src/cmd/utilities.rs [hunk 4]
```diff
@@ -17,7 +17,7 @@ enum CommandOutputType {
     STDERR(Result<String, std::io::Error>),
 }
 
-fn command<P>(binary: P, args: Vec<&str>, envs: Option<Vec<(&str, &str)>>, use_output: bool) -> Command
+fn command<P>(binary: P, args: Vec<&str>, envs: &Vec<(&str, &str)>, use_output: bool) -> Command
```
Comment (erebe) [inline src/cmd/utilities.rs:20]: missing this one

**Pull request merged**